File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    volumn = brown + yellow
    half_length = brown//2 + 2
    
    h = half_length // 2 #세로의 최대 길이
    w = half_length - h
    while 0 < h and h <= w:
        mid = h // 2
        w = half_length - h
        if volumn == w * h:
            answer = [w, h]
            break
        elif volumn < w * h:
            h -= 1
        elif volumn > w * h:
            h += 1
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];


        if (yellow == 1) {
            answer[0] = 3;
            answer[1] = 3;
            return answer;
        } 
        if (yellow == 2) {
            answer[0] = 4;
            answer[1] = 3;
            return answer;
        }

        
       
        
        for (int height = 1; height <= (int)Math.sqrt(yellow); height++) {
            if (yellow%height != 0) continue;
            int width = yellow/height;
            
            if (width * 2 + height * 2 + 4 == brown) {
                answer[0] = width+2;
                answer[1] = height+2;
                break;
            }
        }
       
        return answer;
    }
}